azurelinuxagent/common/utils/processutil.py

```diff
@@ -17,11 +17,58 @@
 # Requires Python 2.6+ and Openssl 1.0+
 #
 
+from azurelinuxagent.common.exception import ExtensionError, ExtensionErrorCodes
 from azurelinuxagent.common.future import ustr
+import os
+import signal
+import subprocess
+import time
 
 TELEMETRY_MESSAGE_MAX_LEN = 3200
 
 
+def wait_for_process_completion_or_timeout(process, timeout):
+    while timeout > 0 and process.poll() is None:
+        time.sleep(1)
+        timeout -= 1
+
+    return_code = None
+
+    if timeout == 0:
+        os.killpg(os.getpgid(process.pid), signal.SIGKILL)
+    else:
+        # process completed or forked; sleep 1 sec to give the child process (if any) a chance to start
```

Comment:
expand this comment to explain why we sleep (previously this was in a private function used in a very specific context so it was easier to understand why)

Reply:
Discussed offline, the comment was already here from before and doesn't change w.r.t refactoring this method out of the `exthandlers` module.